core/src/main/java/hudson/model/Run.java

```diff
@@ -413,9 +413,14 @@ public void addAction(@Nonnull Action a) {
 
     /**
      * Ordering based on build numbers.
+     * If numbers are equal order based on names of parent projects.
      */
     public int compareTo(@Nonnull RunT that) {
-        return this.number - that.number;
+        final int res = this.number - that.number;
+        if (res == 0)
+            return this.getParent().getFullName().compareTo(that.getParent().getFullName());
```

Comment:
It can be done. My main point - it's not safe to have not-aligned `compareTo` and `equals`. Also I can make `Job` `Comparable`, and put that logic there.